### PR TITLE
Add page for CanvasRenderingContext2D.roundRect

### PR DESCRIPTION
**Pull request by yiyix**
Add a page for CanvasRenderingContext2D.roundRect, the whatwg spec is in progress here: https://github.com/whatwg/html/pull/6765

### CODE REVIEW

File: content/en-US/api/roundRect.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  applied to the right.</dd>
+
+  <dt><em>height</em></dt>
+  <dd>A double that represents the rectangle's height. Positive value means the path is drawn
+  clockwise; and negative value meeans it's drawn counterclockwise and flips the rectangle
+  vertically along `x`, that means the radius values that applied to the top corners are now applied
+  to the bottom.</dd>
+
+  <dt><em>radii</em></dt>
+  <dd>Each value `r` in `radii`  could be a number or an object with `{x, y}` properties. If `r` is
```
Comment (mysteryDate) [inline content/en-US/api/roundRect.md]: I feel like we should be consistent between "double" and "number". If we call it "double" above, why not call it "double" here as well?

File: content/en-US/api/roundRect.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  clockwise; and negative value meeans it's drawn counterclockwise and flips the rectangle
+  vertically along `x`, that means the radius values that applied to the top corners are now applied
+  to the bottom.</dd>
+
+  <dt><em>radii</em></dt>
+  <dd>Each value `r` in `radii`  could be a number or an object with `{x, y}` properties. If `r` is 
+  a number, the corresponding corner(s) are drawn as a cicurlar arc with radius `r`; if `r` is an
+  object, the correspondinf corner(s) are drawn as 
+  [elliptical arc](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/ellipse)
+  whose `radiusX` and `radiusY` are equal to `x` and `y`, respectively. `r`, `x` or `y` must be
```
Comment (mysteryDate) [inline content/en-US/api/roundRect.md]: or -> and

File: content/en-US/api/roundRect.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+
+  * If `radii`'s size is 4, then the arc createed by radii[0], radii[1], radii[2] and radii[3]
+  replaces upper left, upper right, lower right and lower left corner of rectangle, respectively.
+
+  * If `radii`'s size is not any of the listed values above, then roundRect aborts and returns an
+  RangeError.
+
+  Note that if the sum of the radii of two corners of the same edge is greater than the length of
+  that edge, all the `radii` of the rounded rectangle are scaled by a factor of length / (r1 + r2).
```
Comment (mysteryDate) [inline content/en-US/api/roundRect.md:65]: Not quite true. If multiple edges are smaller than their respective radii, all radii are scaled by the ratio that corresponds to the edge that requires the most scaling. I think the text here explains it well (https://github.com/whatwg/html/pull/6765)

File: content/en-US/api/roundRect.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+const canvas = document.createElement('canvas');
+canvas.width = 500;
+canvas.height = 500;
+const ctx = canvas.getContext('2d');
+document.body.appendChild(canvas);
+ctx.strokeStyle = '#0f0';
+ctx.lineWidth = 5;
+ctx.roundRect(50, 50, 120, 120, [new DOMPoint(25, 50)]);
+ctx.roundRect(210, 50, 120, 120, [new DOMPoint(20, 50), new DOMPoint(50, 20), new DOMPoint(20, 50), 
+  new DOMPoint(50, 20)]);
```
Comment (mysteryDate) [inline content/en-US/api/roundRect.md:112]: Wanna do some examples with `DOMPointInit`? Also some examples mixing doubles and `DOMPointInit`s e.g.:

`ctx.roundRect(10, 10, 100, 100, [{25, 50}, 10])`

File: content/en-US/api/roundRect.md (hunk 1 above)
Comment (yiyix) [reply content/en-US/api/roundRect.md]: my fault, number is vague. i will remove all usage of number.

File: content/en-US/api/roundRect.md (hunk 2 above)
Comment (yiyix) [reply content/en-US/api/roundRect.md]: done

File: content/en-US/api/roundRect.md (hunk 4 above)
Comment (yiyix) [reply content/en-US/api/roundRect.md:112]: good idea!

File: content/en-US/api/roundRect.md (hunk 3 above)
Comment (yiyix) [reply content/en-US/api/roundRect.md:65]: corrected! I over simplified it.

File: content/en-US/api/roundRect.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+### Parameters
+<dl>
+  <dt><em>x</em></dt>
+  <dd>A double that represents the x-axis (horizontal) coordinate of the rectangle's starting point.</dd>
+
+  <dt><em>y</em></dt>
+  <dd>A double that represents the y-axis (vertical) coordinate of the rectangle's starting point.</dd>
+
+  <dt><em>width</em></dt>
+  <dd>A double that represents the rectangle's width. Positive value means the path is drawn
```
Comment (jpmedley) [inline content/en-US/api/roundRect.md]:
```suggestion
  <dd>A double that represents the rectangle's width. A positive value means the path is drawn
```

File: content/en-US/api/roundRect.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  clockwise; and negative value meeans it's drawn counterclockwise and flips the rectangle
```
Comment (jpmedley) [inline content/en-US/api/roundRect.md]:
```suggestion
  clockwise, and a negative value means it's drawn counterclockwise and flips the rectangle
```

File: content/en-US/api/roundRect.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  horizontally along `y`, that means the radius values that applied to the left corners are now
```
Comment (jpmedley) [inline content/en-US/api/roundRect.md]:
```suggestion
  horizontally along the `y` axis. That means that the radius values that were applied to the left corners are now
```

File: content/en-US/api/roundRect.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+  applied to the right.</dd>
+
+  <dt><em>height</em></dt>
+  <dd>A double that represents the rectangle's height. Positive value means the path is drawn
```
Comment (jpmedley) [inline content/en-US/api/roundRect.md]:
```suggestion
  <dd>A double that represents the rectangle's height. A positive value means the path is drawn
```

File: content/en-US/api/roundRect.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+  clockwise; and negative value meeans it's drawn counterclockwise and flips the rectangle
```
Comment (jpmedley) [inline content/en-US/api/roundRect.md]:
```suggestion
  clockwise, and a negative value means it's drawn counterclockwise and flips the rectangle
```

File: content/en-US/api/roundRect.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+  vertically along `x`, that means the radius values that applied to the top corners are now applied
```
Comment (jpmedley) [inline content/en-US/api/roundRect.md]:
```suggestion
  vertically along the `x` axis. That means the radius values that applied to the top corners are now applied
```

File: content/en-US/api/roundRect.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+  to the bottom.</dd>
+
+  <dt><em>radii</em></dt>
```
Comment (jpmedley) [inline content/en-US/api/roundRect.md:39]: What type of structure is this? An array? A sequence? A list?

We need to rewrite the description to match the grammatical structure of the other arguments. I want to know what this is specifically before I do that.

File: content/en-US/api/roundRect.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+</dd>
+</dl>
+
+## Examples
+
+
+### Drawing roundRect with circular arc
+
+The following example draws 4 rooundRect with `raddi`'s size equals to 1, 2, 3 or 4 and `r` is in
+form of double.
```
Comment (jpmedley) [inline content/en-US/api/roundRect.md]:
```suggestion
The following example draws four `rooundRect` images with `raddi` sizes equal to 1, 2, 3 or 4.  `r` is a double.
```

Comment (jpmedley) [reply content/en-US/api/roundRect.md]: I don't understand the 'r' part. I don't see an 'r' in the example below.

File: content/en-US/api/roundRect.md [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+ctx.roundRect(50, 50, 100, 100, [20]);
+ctx.roundRect(200, 50, 100, 100, [20, 40]);
+ctx.roundRect(50, 200, 100, 100, [10, 25, 40]);
+ctx.roundRect(200, 200, 100, 100, [5, 15, 30, 50]);
+ctx.stroke();
+```
+
+### Drawing roundRect with elliptical arc
+
+The following example draws rooundRect with different elliptical arc.
```
Comment (jpmedley) [inline content/en-US/api/roundRect.md]:
```suggestion
The following example draws a `rooundRect` with a different elliptical arc.
```

**Review by jpmedley — CHANGES_REQUESTED** [review]
This looks pretty good. (Don't let the number of changes fool you. Pro writers don't get off scott free.) I only have two content questions. The rest of typo fixes.

File: content/en-US/api/roundRect.md (hunk 5 above)
Comment (yiyix) [reply content/en-US/api/roundRect.md]: done

File: content/en-US/api/roundRect.md (hunk 11 above)
Comment (yiyix) [reply content/en-US/api/roundRect.md:39]: an array, I rewrite the beginning of the sentence, so it matches the same sentence structure as other items.

File: content/en-US/api/roundRect.md (hunk 12 above)
Comment (yiyix) [reply content/en-US/api/roundRect.md]: ctx.roundRect(50, 50, 100, 100, [20]);
means r = 20, radii = [20]

ctx.roundRect(200, 50, 100, 100, [20, 40]);
means r1 = 20 and r2 =40, radii =  [20, 40]

I will add some comments to explain this.

File: content/en-US/api/roundRect.md (hunk 6 above)
Comment (yiyix) [reply content/en-US/api/roundRect.md]: done

File: content/en-US/api/roundRect.md (hunk 7 above)
Comment (yiyix) [reply content/en-US/api/roundRect.md]: done

File: content/en-US/api/roundRect.md (hunk 8 above)
Comment (yiyix) [reply content/en-US/api/roundRect.md]: done

File: content/en-US/api/roundRect.md (hunk 10 above)
Comment (yiyix) [reply content/en-US/api/roundRect.md]: done

File: content/en-US/api/roundRect.md (hunk 13 above)
Comment (yiyix) [reply content/en-US/api/roundRect.md]: done

**Comment by yiyix** [comment]
Thank you for the review! I updated the spec according the comments.

File: content/en-US/api/roundRect.md (hunk 9 above)
Comment (yiyix) [reply content/en-US/api/roundRect.md]: done

**Review by jpmedley — APPROVED** [review]

**Pull request merged**